aws/resource_aws_cur_report_definition.go

```diff
@@ -0,0 +1,189 @@
+package aws
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/costandusagereportservice"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"log"
+)
+
+func resourceAwsCurReportDefinition() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCurReportDefinitionCreate,
+		Read:   resourceAwsCurReportDefinitionRead,
+		Delete: resourceAwsCurReportDefinitionDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"report_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 256),
+			},
+			"time_unit": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					costandusagereportservice.TimeUnitDaily,
+					costandusagereportservice.TimeUnitHourly,
+				}, false),
+			},
+			"format": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					costandusagereportservice.ReportFormatTextOrcsv}, false),
+			},
+			"compression": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					costandusagereportservice.CompressionFormatGzip,
+					costandusagereportservice.CompressionFormatZip,
+				}, false),
+			},
+			"additional_schema_elements": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						costandusagereportservice.SchemaElementResources,
+					}, false),
+				},
+				Set:      schema.HashString,
+				Required: true,
+				ForceNew: true,
+			},
+			"s3_bucket": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"s3_prefix": {
+				Type:         schema.TypeString,
+				Required:     true,
```

Comment:
If the `s3_prefix` can be left as an empty string, this should be made `Optional: true` and removed from the example in the resource documentation.